### ISSUE TITLE
PNG & PDF rendering are differents on some websites

### ISSUE THREAD

**Issue by Zabrane** [ariya/phantomjs#10623]
_**[zabra...@gmail.com](http://code.google.com/u/107853317813431941706/) commented:**_

> <b>Which version of PhantomJS are you using? Tip: run 'phantomjs --version'.</b>
> 
> v1.5
> 
> <b>What steps will reproduce the problem?</b>
> 1. PNG is perfect (see navigant.png):
> phantomjs rasterize.js http://www.navigant.com/ navigant.png
> 1. PDF is not as expected (see navigant.pdf):
>    phantomjs rasterize.js http://www.navigant.com/ navigant.pdf
> 
> <b>What is the expected output? What do you see instead?</b>
> The PDF and PNG rendering should look the same.
> 
> <b>Which operating system are you using?</b>
> Linux (Fedora 16 and 17), OSX 10.6+.
> 
> <b>Did you use binary PhantomJS or did you compile it from source?</b>
> Both compiled and binary
> 
> <b>Please provide any additional information below.</b>
> Another tool &quot;wkhtmltopdf&quot; (http://code.google.com/p/wkhtmltopdf/) based on Webkit
> is able to correctly generate the PDF (see navigant-wkhtmltopdf.pdf).
> I suspect the that pdf renderingin PhantomJS takes into account css &quot;media&quot; type.
> 
> Thanks for everything

**Disclaimer:**
This issue was migrated on 2013-03-15 from the project's former issue tracker on Google Code, [Issue #623](http://code.google.com/p/phantomjs/issues/detail?id=623).
:star2: &nbsp; **7** people had starred this issue at the time of migration.

**Comment by jorupp** [comment]
It doesn't really solve the issue, but I did come up with something that worked around this for a co-worker.  It's a block of javascript that locks all CSS media queries to their current values using window.matchMedia.  For a site we were trying to export, we were having issues not only with the print stylesheets, but some of the other queries were acting in unexpected ways during the PDF export, and this took care of all the layout issues in a single shot.

Here's the code we added in a page.evaluate block before rendering the PDF in case it helps anyone else in a similar situation: https://gist.github.com/jorupp/2af4d8583bd592b8331f
